src/chunk_constraint.c

```diff
@@ -452,14 +401,17 @@ ChunkConstraints *
 ts_chunk_constraint_scan_by_chunk_id(int32 chunk_id, Size num_constraints_hint, MemoryContext mctx)
 {
 	ChunkConstraints *constraints = ts_chunk_constraints_alloc(num_constraints_hint, mctx);
-	int num_found;
+	ScanKeyData scankey[1];
+	ScanIterator iterator =
+		ts_scan_iterator_create(CHUNK_CONSTRAINT, scankey, AccessShareLock, mctx);
+	int num_found = 0;
 
-	num_found = chunk_constraint_scan_by_chunk_id_internal(chunk_id,
-														   chunk_constraint_tuple_found,
-														   NULL,
-														   constraints,
-														   AccessShareLock,
-														   mctx);
+	scan_by_chunk_id(&iterator.ctx, chunk_id);
```

Comment:
Here there's an implicit dependency on `scankey` and a specific size of it. I think there's a loss in context here that makes this quite non-intuitive.

I you follow my suggestion above to make `ScanKeyData` a fixed-size member of the scan context, you can do all scankey related stuff locally in `scan_by_chunk_id`.

Reply:
Should these `scan_by_` functions just take the iterator pointer instead? See related comment about whether we need to struct-embed `ScannerCtx`